Proofreading edits (#634)
fpvandoorn/carleson: paper/main.tex

To reduce this finitary version of the Carleson theorem to the discrete version, we make divisions in both the spatial and frequency domains. The spatial discretization will be achieved by a \emph{grid structure}, after which a subordinate frequency discretization is encoded by a \emph{tile structure}.

> A \emph{grid structure} on $X$ is a triple $(\mathcal{D}, c, s)$ consisting of a finite set $\mathcal{D}$ of so-called dyadic cubes, a function $c\colon \mathcal{D} \to X$ called the center function, and a surjective function $s\colon \mathcal{D} \to [-S, S]$ called the scale function, that together satisfy the conditions \ref{it:coverdyadic}-\ref{it:eq-small-boundary} below. Here, a dyadic cube is just a pair $(I, k)$ of a Borel set $I$ in $X$ and an integer $k \in [-S, S]$. Usually we abuse notation and write $I$ for a dyadic cube $(I, k)$.

Reviewer: Pre-existing, can happen in a follow-up: add an external link to "grid structure"?

Author: I checked Pietro's PR, it does not add this specific link so I added it.

    fourierCoeffOn Real.two_pi_pos f n * fourier n (x : AddCircle (2 * π))
\end{lstlisting}

> In contrast to the classical result, to be able to formalize the statements of the metric and linearized versions of Carleson's theorem, we first needed to formalize a good amount of new concepts, starting with the definition of doubling metric measure space (see Definition~\ref{def:doubling}), which we denote by \lean{MeasureTheory.DoublingMeasure}\href{https://github.com/fpvandoorn/carleson/blob/f2f614db47f1a30e3fceaa9c2e1f99ff356dd577/Carleson/Defs.lean#L87}{\extlink}. The constant $A$ appearing in this definition will typically be $2^a$.

Reviewer: This sentence is getting long, let's split it.
```suggestion
In contrast to the classical result, to be able to formalize the statements of the metric and linearized versions of Carleson's theorem, we first needed to formalize a good amount of new concepts.
The first such conept is a doubling metric measure space (see Definition~\ref{def:doubling}), which we denote by \lean{MeasureTheory.DoublingMeasure}\href{https://github.com/fpvandoorn/carleson/blob/f2f614db47f1a30e3fceaa9c2e1f99ff356dd577/Carleson/Defs.lean#L87}{\extlink}. The constant $A$ appearing in this definition will typically be $2^a$.
```

%\end{lstlisting}

The \lean{carlesonOperator}\href{https://github.com/fpvandoorn/carleson/blob/f2f614db47f1a30e3fceaa9c2e1f99ff356dd577/Carleson/Defs.lean#L218}{\extlink} is defined as a special case of the

> \lean{linearizedCarlesonOperator}\href{https://github.com/fpvandoorn/carleson/blob/f2f614db47f1a30e3fceaa9c2e1f99ff356dd577/Carleson/Defs.lean#L212}{\extlink}; these definitions correspond to those introduced in equations \eqref{def-main-op} and \eqref{def-T-r}, respectively. The expression $\mlean{\|carlesonOperatorIntegrand K (Q x) R_1 R_2 f x\|_e}$ denotes the extended norm of the integral appearing in Equation \eqref{def-T-r}, taking values in $\ennreal$ (see \ref{subsec:enorm} for more details). % JMS the \mlean command does not render correct, don't have a clear fix

Reviewer: Good catch!
```suggestion
\lean{linearizedCarlesonOperator}\href{https://github.com/fpvandoorn/carleson/blob/f2f614db47f1a30e3fceaa9c2e1f99ff356dd577/Carleson/Defs.lean#L212}{\extlink}; these definitions correspond to those introduced in equations \eqref{def-main-op} and \eqref{def-T-r}, respectively. The expression $\mlean{\|carlesonOperatorIntegrand K (Q x) R_1 R_2 f x\|_e}$ denotes the extended norm of the integral appearing in Equation \eqref{def-T-r}, taking values in $\ennreal$ (see \ref{subsec:enorm} for more details).
% JMS the \mlean command does not render correctly (spaces are not rendered)
```

\subsection{The blueprint writing process}\label{subsec:blueprint_writing}

As mentioned in \S\ref{sec:org}, the blueprint~\cite{CarlesonBlueprint} for the Carleson formalization project was written
by the harmonic analysis group in Bonn. When the formalization project was announced, it was not finished in its entirety: some parts of Section~7,
which deals with arguably the most complicated case of the proof of \Cref{discrete-Carleson},
were still being finalized, and

> Section~3, which reduces \Cref{metric-space-Carleson} to \Cref{finitary-Carleson},

Reviewer: The blueprint uses \section (and this is on the review checklist anyway), so I'll take it!

In the end, we decided to prove that this function was bounded with compact support, and additionally bounded everywhere.

These classes are also tagged with the \lean{@[fun_prop]} attribute.

> \lean{fun_prop}\footnote{\url{https://leanprover-community.github.io/mathlib4_docs/Mathlib/Tactic/FunProp.html}} is a \mathlib tactic developed by Tomáš Skřivan

Reviewer: Good idea to link fun_prop: I think we should link https://leanprover-community.github.io/mathlib4_docs/tactics.html#Mathlib.Meta.FunProp.funPropTacStx (or a more stable link, e.g. pinned to a recent version of mathlib) instead.

Author: Sure, done.

In the end, we decided to prove that this function was bounded with compact support, and additionally bounded everywhere.

These classes are also tagged with the \lean{@[fun_prop]} attribute.

> \lean{fun_prop}\footnote{\url{https://leanprover-community.github.io/mathlib4_docs/tactics.html#Mathlib.Meta.FunProp.funPropTacStx}} is a \mathlib tactic developed by Tomáš Skřivan

Reviewer: I think this will fix the compilation error:
```suggestion
\lean{fun_prop}\footnote{\url{https://leanprover-community.github.io/mathlib4_docs/tactics.html\#Mathlib.Meta.FunProp.funPropTacStx}} is a \mathlib tactic developed by Tomáš Skřivan
```
